testsuite/integration/basic/src/test/java/org/wildfly/test/integration/observability/opentelemetry/BasicOpenTelemetryTestCase.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2021 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.test.integration.observability.opentelemetry;
+
+import javax.inject.Inject;
+
+import io.opentelemetry.api.OpenTelemetry;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import io.opentelemetry.api.trace.Tracer;
+
+@RunWith(Arquillian.class)
+public class BasicOpenTelemetryTestCase {
```

Comment:
This test case covers the DUP. I would expect here also a real integration with a collector.

Reply:
That would be ideal, but, as written, the extension doesn't support the kind of extensibility it would take to do that. The OpenTelemetry instance is created once and shared across all deployments. The pieces that would need to be swapped out in order to capture/process/verify the spans in a test are not changeable once the otel instance has been created. In order to support that, we would need to modify when/where the otel instance is created, as well as providing a mechanism for the application to provide (e.g., via CDI) alternative implementations for the various pieces that wired together inside the otel instance. All of that is doable, of course, but I chose NOT to support that in this first go in order to keep things simple, waiting to add that kind of support if/when users request it. We can do that NOW, though that might impact the schedule (hard to say). All that said, I _have_ tested manually with a Jaeger instance running locally and have seen the traces exported as expected. That is not as good as automated regression testing, of course, but it's at least a bit better than nothing. :) If you (or anyone) feels strongly that we should implement the test like this, please speak up. I may try to get started regardless and see what happens.

Reply:
I discussed this with @bstansberry. We won't be expanding the RFE so that we can run a test case. If/when we do add this functionality, this kind of test can and will be added.

Reply:
I spoke too quickly. There might be a way to test this pretty simply with the Jaeger collector. I'll get back to you.